### PR TITLE
Gateway validation error message

### PR DESCRIPTION
**Pull request by werdes72**
More descriptive error.

### CODE REVIEW

File: apis/gateway/v2alpha1/apirule_types.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ type APIRuleSpec struct {
 	Service *Service `json:"service,omitempty"`
 	// Specifies the Istio Gateway to be used.
 	// +kubebuilder:validation:MaxLength=127
-	// +kubebuilder:validation:XValidation:rule=`self.matches('^[a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?/([a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?)$')`,message="Gateway is not valid"
+	// +kubebuilder:validation:XValidation:rule=`self.matches('^[a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?/([a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?)$')`,message="Gateway should be in namespace/name format"
```
Comment (videlov) [inline apis/gateway/v2alpha1/apirule_types.go:46]: I would call it a `must` not only `should`

**Review by videlov — DISMISSED** [review]

**Comment by videlov** [comment]
@nataliasitko is the error message okay this way ?

File: config/crd/bases/gateway.kyma-project.io_apirules.yaml [hunk 2]
```diff
@@ -358,7 +358,7 @@ spec:
                 maxLength: 127
                 type: string
                 x-kubernetes-validations:
-                - message: Gateway is not valid
+                - message: Gateway must be in namespace/name format
```
Comment (nataliasitko) [inline config/crd/bases/gateway.kyma-project.io_apirules.yaml:361]:
```suggestion
                - message: Gateway must be in the namespace/name format
```

File: controllers/gateway/api_controller_integration_test.go [hunk 3]
```diff
@@ -1433,7 +1433,7 @@ var _ = Describe("APIRule Controller", Serial, func() {
 				serviceTeardown(svc)
 			}()
 			Expect(err).Should(HaveOccurred())
-			Expect(err.Error()).To(ContainSubstring("spec.gateway: Invalid value: \"string\": Gateway is not valid"))
+			Expect(err.Error()).To(ContainSubstring("spec.gateway: Invalid value: \"string\": Gateway must be in namespace/name format"))
```
Comment (nataliasitko) [inline controllers/gateway/api_controller_integration_test.go:1436]:
```suggestion
			Expect(err.Error()).To(ContainSubstring("spec.gateway: Invalid value: \"string\": Gateway must be in the namespace/name format"))
```

File: apis/gateway/v2alpha1/apirule_types.go [hunk 4]
```diff
@@ -43,7 +43,7 @@ type APIRuleSpec struct {
 	Service *Service `json:"service,omitempty"`
 	// Specifies the Istio Gateway to be used.
 	// +kubebuilder:validation:MaxLength=127
-	// +kubebuilder:validation:XValidation:rule=`self.matches('^[a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?/([a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?)$')`,message="Gateway is not valid"
+	// +kubebuilder:validation:XValidation:rule=`self.matches('^[a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?/([a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?)$')`,message="Gateway must be in namespace/name format"
```
Comment (nataliasitko) [inline apis/gateway/v2alpha1/apirule_types.go:46]:
```suggestion
	// +kubebuilder:validation:XValidation:rule=`self.matches('^[a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?/([a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?)$')`,message="Gateway must be in the namespace/name format"
```

**Review by videlov — APPROVED** [review]

**Pull request merged**